MapleServer2/PacketHandlers/Game/UserSyncHandler.cs

```diff
@@ -92,9 +99,9 @@ private static bool IsOutOfBounds(CoordF coord, CoordS[] boundingBox)
             return false;
         }
 
-        private static bool IsCoordSafe(GameSession session, CoordF coord)
+        private static bool IsCoordSafe(GameSession session, CoordS currentCoord, CoordF closestCoord)
         {
-            return (session.Player.SafeCoord - coord).Length() > 200 && session.FieldPlayer.Coord.Z == coord.Z && !session.Player.OnAirMount; // Save last coord if player is not falling and not in a air mount
+            return (session.Player.SafeBlock - closestCoord).Length() > 350 && session.FieldPlayer.Coord.Z == currentCoord.Z && !session.Player.OnAirMount && MapMetadataStorage.BlockExists(session.Player.MapId, closestCoord.ToShort()); // Save last coord if player is not falling and not in a air mount
```

Comment:
```suggestion
            // Save last coord if player is not falling and not in a air mount
            return MapMetadataStorage.BlockExists(session.Player.MapId, closestCoord.ToShort()) &&
                !session.Player.OnAirMount &&
                (session.Player.SafeBlock - closestCoord).Length() > 350 &&
                session.FieldPlayer.Coord.Z == currentCoord.Z;
```

I'd check if the block exists and the player is not on an air mount.

Reply:
@angelotadeucci note that every check is on a new line